internal/internal_workflow_client.go

```diff
@@ -1516,6 +1520,14 @@ func (w *workflowClientInterceptor) ExecuteWorkflow(
 		Header:                   header,
 	}
 
+	var eagerExecutor *eagerWorkflowExecutor
+	if in.Options.EnableEagerStart && w.client.capabilities.GetEagerWorkflowStart() && w.eagerDispatcher != nil {
+		eagerExecutor = w.eagerDispatcher.applyToRequest(startRequest)
+		if eagerExecutor != nil {
+			defer eagerExecutor.release()
+		}
+	}
```

Comment:
I am not wondering if there is value in a separate dispatcher and executor concept since this is so straightforward and done on only this call. I wonder if you can just store a `eagerWorkflowWorkers sync.Map` on `WorkflowClient` and change this to:

```go
var eagerWorkerInUse eagerWorker
if in.Options.EnableEagerStart && w.client.capabilities.GetEagerWorkflowStart() {
	eagerWorkerIface, _ := w.client.eagerWorkflowWorkers.Load(request.GetTaskQueue().GetName())
	eagerWorkerInUse, _ = eagerWorkerIface.(eagerWorker)
	request.RequestEagerExecution = eagerWorkerInUse != nil && eagerWorkerInUse.tryReserveSlot()
}
```

And after response:

```go
if request.RequestEagerExecution {
	if task := response.GetEagerWorkflowTask(); task != nil {
		eagerWorkerInUse.processTaskAsync(&eagerWorkflowTask{task}, eagerWorkerInUse.releaseSlot)
	} else {
		eagerWorkerInUse.releaseSlot()
	}
}
```

Reply:
I prefer the abstraction of a separate dispatcher and executor